pkg/kds/context/context.go

```diff
@@ -353,14 +355,8 @@ func SkipUnsupportedHostnameGenerator(ctx context.Context, clusterID string, fea
 	return true
 }
 
+// ZoneProvidedFilter filters resources that should be sent from Zone to Global
+// a resource is sent from zone to global only if it was created in this zone.
 func ZoneProvidedFilter(_ context.Context, localZone string, _ kds.Features, r core_model.Resource) bool {
-	if zi, ok := r.(*core_mesh.ZoneIngressResource); ok {
-		// Old zones don't have a 'kuma.io/zone' label on ZoneIngress, when upgrading to the new 2.6 version
-		// we don't want Zone CP to sync ZoneIngresses without 'kuma.io/zone' label to Global pretending
-		// they're originating here. That's why upgrade from 2.5 to 2.6 (and 2.7) requires casting resource
-		// to *core_mesh.ZoneIngressResource and checking its 'spec.zone' field.
-		// todo: remove in 2 releases after 2.6.x
-		return !zi.IsRemoteIngress(localZone)
-	}
-	return core_model.IsLocallyOriginated(config_core.Zone, r.GetMeta().GetLabels()) || r.Descriptor().KDSFlags == core_model.ZoneToGlobalFlag
+	return core_model.IsLocallyOriginated(config_core.Zone, r.GetMeta().GetLabels())
```

Comment:
Don't we still need to check the KDSFlag here?

Reply:
Nope because we only pass resources that use KDS in the filter already: https://github.com/kumahq/kuma/blob/4ab84acd60a6921909c287444c2276071dcc24ed/pkg/kds/context/context.go#L47-L48

so checking here was just redundant, on top of this we were using `||` which probably means we were sending back up stuff without origin labels which is not longer possible